### PR TITLE
Simple faceting implementation for fast numerical fields

### PR DESCRIPTION
**Pull request by Fiedzia**

### CODE REVIEW

**Comment by fulmicoton** [comment]
Thanks. 

Im a bit swamped at the moment but I should have to hit this PR time later this week.

At first glance it looks good.

**Comment by lnicola** [comment]
Can you extend this to `i64` fields? Looks good otherwise.

**Comment by Fiedzia** [comment]
Done

**Comment by lnicola** [comment]
Thanks, looks good to me.

You can fix the CI error by running `cargo fmt` with `rustfmt 0.8.4`, but it's no biggie.

**Comment by lnicola** [comment]
I squashed the commits, rebased against `master` and pusehd to https://github.com/tantivy-search/tantivy/tree/facets.

**Review by celaus — CHANGES_REQUESTED** [review]
Thanks for the PR, I think it looks good, aside from two minor things 👍

**Comment by fulmicoton** [comment]
It was merged. Because of some commit squashing I assume, this branch was not automatically closed.

I had to do a bunch of modification for the following reasons.

- `.new()` was not called
- I removed the wrapper that makes it possible to have more than once facet. `tantivy` already has
`chain()` to handle multiple collectors with a static dispatch.
- `set_segment()` was not handled.

Thanks for your PR !

File: src/collector/facet_collector.rs [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    }
+
+    fn collect(&mut self, doc: DocId, _: Score) {
+        let val = self.ff_reader.as_ref().unwrap().get(doc);
+        *(self.counters.entry(val).or_insert(0)) += 1;
+    }
+} 
+
+
+enum FacedType {
```
[the remark this answers is not in the document]
Comment (Fiedzia) [reply src/collector/facet_collector.rs:62]: Yes, it's just a typo

**Pull request closed without merging**